app/pages/project/project-page.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+import { Markdown } from 'markdownz';
+import { sugarClient } from 'panoptes-client/lib/sugar';
+import ProjectNavbar from './project-navbar';
+import PotentialFieldGuide from './potential-field-guide';
+
+export default class ProjectPage extends React.Component {
+  constructor() {
+    super();
+    const lastSugarSubscribedID = null;
+  }
+
+  getChildContext() {
+    return this.context.geordi;
+  }
+
+  componentDidMount() {
+    document.documentElement.classList.add('on-project-page');
+    this.updateSugarSubscription(this.props.project);
+    (this.context.geordi) ? this.context.geordi.remember({ projectToken: this.props.project.slug }) : null;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.project !== this.props.project) {
+      this.updateSugarSubscription(nextProps.project);
+      (this.context.geordi) ? this.context.geordi.remember({ projectToken: nextProps.project.slug }) : null;
+    }
+  }
+
+  componentWillUnmount() {
+    document.documentElement.classList.remove('on-project-page');
+    this.updateSugarSubscription(null);
+    this.context.geordi.forget(['projectToken']);
+  }
+
+  updateSugarSubscription(project) {
+    if (this.lastSugarSubscribedID) {
+      sugarClient.unsubscribeFrom(`project-${this.lastSugarSubscribedID}`);
+    }
+    if (project) {
+      sugarClient.subscribeTo(`project-${this.lastSugarSubscribedID}`);
+    }
+  }
+
+  render() {
+    const projectPath = `/projects/${this.props.project.slug}`;
+    const onHomePage = this.props.routes[2].path === undefined;
+    const pages = this.props.pages.reduce((map, page) => {
+      map[page.url_key] = page;
+      return map;
+    }, {});
+
+    const renderAnnouncement = (this.props.project.configuration && !!this.props.project.configuration.announcement) ?
+      <div className="informational project-announcement-banner">
+        <Markdown>
+          {this.props.project.configuration.announcement}
+        </Markdown>
+      </div> :
+      null;
+
+    const renderNavbar = (!onHomePage) ?
+      <div>
```

Comment:
Is this pattern preferable to `{!!someFlag && <SomeComponent/>}` which we use elsewhere for conditional rendering in JSX?

Reply:
It's just what I got used doing. It's quite common to see it online, but I suppose it just boils down to what reads better. I find it easier than your snippet, but I'm happy to change it if you want.

Reply:
I've no strong opinion on this one, as long as the variables are renamed here.

Here's a real example of the pattern I was thinking of:
https://github.com/zooniverse/Panoptes-Front-End/blob/2ae4fd54e34de274a3ddbae32eeeec5bfff5ae62/app/classifier/task.jsx#L65-L80